# HG changeset patch
# User Rik <rik@octave.org>
# Date 1515720413 28800
#      Thu Jan 11 17:26:53 2018 -0800
# Node ID 9b36fc7bd0026e7a1c377bae98b79da44a60759e
# Parent  c12b5cec5c6c57c84206af7a4f37d255ed4ebd37
__add_default_menu__.m: Don't close the wrong figure if File->Close used in FLTK.

* __add_default_menu__.m: Connect File->Close to close_cb() callback.
* __add_default_menu__.m (close_cb): Callback which uses closes gcbf, rather
than gcf.

diff --git a/scripts/plot/util/private/__add_default_menu__.m b/scripts/plot/util/private/__add_default_menu__.m
--- a/scripts/plot/util/private/__add_default_menu__.m
+++ b/scripts/plot/util/private/__add_default_menu__.m
@@ -34,17 +34,17 @@ function __add_default_menu__ (fig)
   endif
 
   obj = findall (fig, "-depth", 1, "tag", "__default_menu__", "label", "&File");
   if (isempty (obj))
     __f = uimenu (fig, "label", "&File", "handlevisibility", "off",
                        "tag", "__default_menu__");
       uimenu (__f, "label", "&Save", "callback", @save_cb);
       uimenu (__f, "label", "Save &As", "callback", @save_cb);
-      uimenu (__f, "label", "&Close", "callback", "close (gcf)");
+      uimenu (__f, "label", "&Close", "callback", @close_cb);
 
     __e = uimenu (fig, "label", "&Edit", "handlevisibility", "off",
                        "tag", "__default_menu__");
       uimenu (__e, "label", "Toggle &grid on all axes", "tag", "toggle", "callback", @grid_cb);
       uimenu (__e, "label", "Show grid on all axes", "tag", "on", "callback", @grid_cb);
       uimenu (__e, "label", "Hide grid on all axes", "tag", "off", "callback", @grid_cb);
       uimenu (__e, "label", "Auto&scale all axes", "callback", @autoscale_cb);
       gm = uimenu (__e, "label", "GUI &Mode (on all axes)");
@@ -92,16 +92,21 @@ function __save_as__ (caller)
       hgsave (caller, fname);
     else
       saveas (caller, fname);
     endif
   endif
 endfunction
 
 
+function close_cb (h, e)
+  close (gcbf);
+endfunction
+
+
 function [hax, fig] = __get_axes__ (h)
   ## Get parent figure
   fig = ancestor (h, "figure");
 
   ## Find all axes which aren't legends
   hax = findobj (fig, "type", "axes", "-not", "tag", "legend");
 endfunction
 
